### PR TITLE
Social Media Logins and ASP .NET CORE upgrade

### PR DESCRIPTION
**Pull request by logophobia**
- Contains a migration to asp net core 2
- Added logins for google/facebook/twitter
- Revised structure of some controllers to resemble the way the "reference" application does things in asp net core 2
- Remove some unused code surrounding accounts. Probably not a good idea to leave active account related code that can't be reached/tested, security/efficiency wise

This will merge into master, I'll make a separate PR that merges into the friesland branch w/ the correct translations and layout.

### CODE REVIEW

File: CollAction/Program.cs [hunk 1]
```diff
@@ -3,22 +3,24 @@
...
 using System.Linq;
 using System.Threading.Tasks;
+using CollAction;
+using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
 
-namespace CollAction
+namespace WebApplication2
```
Comment (ljwboschker) [inline CollAction/Program.cs]: Strange namespace...

File: CollAction/Startup.cs [hunk 2]
```diff
@@ -67,6 +67,23 @@ public void ConfigureServices(IServiceCollection services)
                 options.Password.RequiredLength = 8;
             });
 
+            services.AddAuthentication()
+                    .AddFacebook(options =>
+                    {
+                        options.AppId = Configuration["Authentication:Facebook:AppId"];
```
Comment (ljwboschker) [inline CollAction/Startup.cs]: Where are these values coming from? I couldn't find them in the values. Should they be in env_collaction?

Comment (ljwboschker) [reply CollAction/Startup.cs]: Also, I think the keys, and the accounts you can use to manage them, should be stored in the password-tool.

File: CollAction/Views/Account/ExternalLogin.cshtml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<div id="login-starting">
+    <div class="container">
+        <div class="row">
+            <div class="col-xs-12">
+                <h1>@ViewData["Title"]</h1>
+                <h4>Associate your @ViewData["LoginProvider"] account.</h4>
+                <p class="text-info">
+                    You've successfully authenticated with <strong>@ViewData["LoginProvider"]</strong>.
+                    Please enter an email address for this site below and click the Register button to finish
+                    logging in.
```
Comment (ljwboschker) [inline CollAction/Views/Account/ExternalLogin.cshtml]: The text could be less technical, more focussed on the user. Perhaps we can ask UX.

File: CollAction/Program.cs [hunk 4]
```diff
@@ -3,13 +3,12 @@
...
 using System.Linq;
 using System.Threading.Tasks;
-using CollAction;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 
-namespace WebApplication2
+namespace CollAction
```
Comment (ljwboschker) [inline CollAction/Program.cs]: Ah. Forget my previous comment about this :-)

File: CollAction/Program.cs (hunk 1 above)
Comment (logophobia) [reply CollAction/Program.cs]: This was corrected in a later commit. You can look at the "Changes" tab for the effective changes.

File: CollAction/Startup.cs (hunk 2 above)
Comment (logophobia) [reply CollAction/Startup.cs]: They are stored in the same place as all the other keys, safely. Either in aws configuration, or the password tool.

But you're right, I should add them to the env_collaction.template as well, for easier configuration.

File: CollAction/Views/Account/ExternalLogin.cshtml (hunk 3 above)
Comment (logophobia) [reply CollAction/Views/Account/ExternalLogin.cshtml]: I'll ask in slack

**Comment by logophobia** [comment]
First I'll merge the english version with master (it's based on master since it's been in development so long), then I'll merge a dutch version with the friesland branch. Easier that way.

Also, changed the text and added the configuration options to the template.

**Pull request merged**